### ISSUE TITLE
Configuration#environment not always converted to symbol

### ISSUE THREAD

**Issue by replaid** [braintree/braintree_ruby#148]
### General information

* SDK/Library version: At least since 2.72.0 through current master
* Environment: all
* Language, language version, and OS: Ruby 2.2.4p230, though it's not environment-specific

### Issue description

Braintree::Configuration converts environment values to symbols when using the technique of setting defaults with class instance variables and calling `.instantiate`:

```ruby
> Braintree::Configuration.environment='production'
=> "production"
> Braintree::Configuration.merchant_id='mid'
=> "mid"
> Braintree::Configuration.private_key='privkey'
=> "privkey"
> Braintree::Configuration.public_key='pubkey'
=> "pubkey"
> bt_instantiated = Braintree::Configuration.instantiate
=> #<Braintree::Configuration:0x007fa27bb7ff28 @custom_user_agent=nil, @endpoint=nil, @http_open_timeout=60, @http_read_timeout=60, @logger=#<Logger:0x007fa27ab38138 @progname=nil, @level=1, @default_formatter=#<Logger::Formatter:0x007fa27ab38110 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x007fa27ab380c0 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDOUT>>, @mutex=#<Logger::LogDevice::LogDeviceMutex:0x007fa27ab38098 @mon_owner=nil, @mon_count=0, @mon_mutex=#<Mutex:0x007fa27ab38048>>>>, @merchant_id="mid", @public_key="pubkey", @private_key="[FILTERED]", 
# @environment=:production, 
# @proxy_address=nil, @proxy_port=nil, @proxy_user=nil, @proxy_pass=nil, @ssl_version=nil>
> bt_instantiated.environment
=> :production
```

But not when setting values while calling the constructor directly—it leaves string values as strings:

```ruby
> bt1 = Braintree::Configuration.new(environment: 'production')
=> #<Braintree::Configuration:0x007fa280603d10 @custom_user_agent=nil, @endpoint=nil, @http_open_timeout=nil, @http_read_timeout=nil, @logger=nil, @merchant_id=nil, @public_key=nil, @private_key=nil, 
# @environment="production",
# @proxy_address=nil, @proxy_port=nil, @proxy_user=nil, @proxy_pass=nil, @ssl_version=nil>
> bt1.environment
=> "production"
```

It would be helpful to do the conversion and validation done in `Configuration.environment=` whenever the environment is set.

**Comment by crookedneighbor** [comment]
You're right that it is inconsistent, but we can't make this change without a major version bump, as developers may be relying on the current behavior. 

We'll put this on our list to consider when doing a major version bump.

**Comment by mobitar** [comment]
Came here to say the same thing. I'm loading these values from a .env file like so:

``` ruby
Braintree::Configuration.environment = ENV['BRAINTREE_ENV']
Braintree::Configuration.merchant_id = ENV['BRAINTREE_MERCHANT_ID']
```

However this does not work and does not throw any errors other than after I check out, and get a "merchant network unavailable error".

I even tried `ENV['BRAINTREE_ENV'].to_sym` and `ENV['BRAINTREE_ENV'].intern`, but neither of those worked.

I ended up having to do:

``` ruby
Braintree::Configuration.environment = Rails.env.production? ? :production : :sandbox
```

Might be helpful to catch this early on and throw an error if the value is not in the format the library expects.

**Comment by crookedneighbor** [comment]
Should be simple enough to automatically convert the string value passed in (`"sandbox"` or `"production"`) to a symbol internally in the SDK. That should prevent this from being a breaking change.

Will update here when it's ready.

**Comment by jackellenberger** [comment]
This has been resolved with the release of [2.91.0](https://github.com/braintree/braintree_ruby/commit/a2e505af16b90de17e35b4d7e52420e94153ca7c#diff-0345291ca95257b0c4400a5cdcadf7c6R118).
